storage/innobase/dict/dict0stats.cc

```diff
@@ -1629,25 +1709,25 @@ dict_stats_analyze_index_level(
 	entered at all and there is one page in the whole tree which is
 	empty or the loop was entered but this is level 0, contains one page
 	and all records are delete-marked */
-	if (*total_pages == 0) {
+	if (stats->total_pages == 0) {
 
 		ut_ad(level == 0);
-		ut_ad(*total_recs == 0);
+		ut_ad(stats->total_recs == 0);
 
-		*total_pages = 1;
+		stats->total_pages = 1;
 	}
 
 	/* if there are records on this level and boundaries
 	should be saved */
-	if (*total_recs > 0 && n_diff_boundaries != NULL) {
+	if (stats->total_recs > 0 && n_diff_boundaries != NULL) {
 
 		/* remember the index of the last record on the level as the
 		last one from the last group of equal keys; this holds for
 		all possible prefixes */
 		for (i = 0; i < n_uniq; i++) {
 			ib_uint64_t	idx;
 
-			idx = *total_recs - 1;
+			idx = stats->total_recs - 1;
 
 			n_diff_boundaries[i].push_back(idx);
 		}
```

Comment:
Do we really want to reuse the same variable `i` in each loop? This could be written shorter as
```c++
for (ulint i = 0; i < n_uniq; i++) {
	n_diff_boundaries[i].emplace_back(stats->total_recs - 1);
}
```
(or even shorter if we omit the redundant `total_` prefix).